src/interface.jl

```diff
@@ -0,0 +1,252 @@
+    Tracker.gradient(z -> transform(b, z), y)[1]
+end
+function jacobian(b::Inversed{<: ADBijector{<: TrackerAD}}, y::AbstractVector{<: Real})
+    Tracker.jacobian(z -> transform(b, z), y)
+end
+
+# TODO: allow batch-computation, especially for univariate case?
+"Computes the absolute determinant of the Jacobian of the inverse-transformation."
+function logabsdetjac(b::ADBijector, x::Real)
+    log(abs(jacobian(b, x)))
+end
+function logabsdetjac(b::ADBijector, x::AbstractVector{<:Real})
+    fact = lu(jacobian(b, x), check=false)
+    issuccess(fact) ? log(abs(det(fact))) : -Inf # TODO: or smallest possible float?
+end
+
+###############
+# Composition #
+###############
+
+struct Composed{A} <: Bijector
+    ts::A
+end
+
+compose(ts...) = Composed(ts)
+
+# The transformation of `Composed` applies functions left-to-right
+# but in mathematics we usually go from right-to-left; this reversal ensures that
+# when we use the mathematical composition ∘ we get the expected behavior.
+# TODO: change behavior of `transform` of `Composed`?
+∘(b1::B1, b2::B2) where {B1 <: Bijector, B2 <: Bijector} = compose(b2, b1)
+
+inv(ct::Composed) = Composed(map(inv, reverse(ct.ts)))
+
+# TODO: can we implement this recursively, and with aggressive inlining, make this type-stable?
+function transform(cb::Composed, x)
+    res = x
+    for b ∈ cb.ts
+        res = transform(b, res)
+    end
+
+    return res
+end
+
+(cb::Composed)(x) = transform(cb, x)
+
+function forward(cb::Composed, x)
+    res = (rv=x, logabsdetjac=0)
+    for t in cb.ts
+        res′ = forward(t, res.rv)
+        res = (rv=res′.rv, logabsdetjac=res.logabsdetjac + res′.logabsdetjac)
+    end
+    return res
+end
+
+##############################
+# Example bijector: Identity #
+##############################
+
+struct Identity <: Bijector end
+transform(::Identity, x) = x
+transform(::Inversed{Identity}, y) = y
+(b::Identity)(x) = transform(b, x)
+
+forward(::Identity, x) = (rv=x, logabsdetjac=zero(x))
+
+logabsdetjac(::Identity, y::T) where T <: Real = zero(T)
+logabsdetjac(::Identity, y::AbstractVector{T}) where T <: Real = zero(T)
+
+const IdentityBijector = Identity()
+
+###############################
+# Example: Logit and Logistic #
+###############################
+using StatsFuns: logit, logistic
+
+struct Logit{T<:Real} <: Bijector
+    a::T
+    b::T
+end
+
+transform(b::Logit, x::Real) = logit((x - b.a) / (b.b - b.a))
+transform(ib::Inversed{Logit{T}}, y::Real) where T <: Real = (ib.orig.b - ib.orig.a) * logistic(y) + ib.orig.a
+(b::Logit)(x) = transform(b, x)
+
+logabsdetjac(b::Logit{<:Real}, x::Real) = log((x - b.a) * (b.b - x) / (b.b - b.a))
+forward(b::Logit, x::Real) = (rv=transform(b, x), logabsdetjac=-logabsdetjac(b, x))
+
+
+#######################################################
+# Constrained to unconstrained distribution bijectors #
+#######################################################
+struct DistributionBijector{AD, D} <: ADBijector{AD} where D <: Distribution
```

Comment:
Can re-write this more compactly:
```julia
struct DistributionBijector{AD, <:Distribution} <: ADBijector{AD}
```

Reply:
We need the type for the `dist::D` field, so I don't think we can

Reply:
You're totally right. What we could do though is
```julia
struct DistributionBijector{AD, D<:Distribution} <: ADBijector{AD}
```